src/fmllauncher/java/net/minecraftforge/fml/loading/ModJarURLHandler.java

```diff
@@ -47,21 +48,25 @@ protected URLConnection openConnection(URL url) {
         private Path resource;
         private String modpath;
         private String modid;
-        private Optional<Manifest> manifest;
+        private Manifest manifest;
 
         public ModJarURLConnection(final URL url) {
             super(url);
         }
 
         @Override
-        public void connect()
+        public void connect() throws IOException
         {
             if (resource == null) {
                 modid = url.getHost();
                 // trim first char
                 modpath = url.getPath().substring(1);
-                resource = FMLLoader.getLoadingModList().getModFileById(modid).getFile().findResource(modpath);
-                manifest = FMLLoader.getLoadingModList().getModFileById(modid).getManifest();
+                ModFileInfo modFileInfo = FMLLoader.getLoadingModList().getModFileById(modid);
+                if (modFileInfo == null) {
+                    throw new IOException("No mod file found for " + modid);
+                }
+                resource = modFileInfo.getFile().findResource(modpath);
+                manifest = modFileInfo.getManifest().orElse(null);
```

Comment:
Why do you unpack this then just repack it as an optional in the getter? Does calling `orElse` possibly throw an IOException? I feel like a comment is needed here explaining this if nothing else.

Since we return an optional later anyways, it may make sense to repack the return here and keep the field type the same